packages/cli/src/scaling/scaling.service.ts

```diff
@@ -234,6 +233,18 @@ export class ScalingService {
 		});
 	}
 
+	/**
+	 * Register listeners on a `main` process for Bull queue events.
+	 */
+	private registerMainListeners() {
+		this.queue.on('global:progress', (_jobId: JobId, msg: MessageToMain) => {
```

Comment:
I guess the type for `msg` is not correct, or otherwise we wouldn't need the `if (msg.kind === ... )`. Now I don't know if there are other events that might be emitted, but we could either

1. Type the `msg` as `unknown` and have a type guard that checks based on `kind` if the message is a `RespondToWebhookMessage` (i.e. there might be other messages as well)
2. Type the `msg` as `RespondToWebhookMessage` and `assert(msg.kind === ...)` (i.e. there are no other message types)